Improve Touch (Part 2 - Multitouch)

rikner opened the pull request:
## Motivation (current vs expected behavior)

Adds support for multitouch

Currently the branch causes huge slowdowns with multitouch, we may need to reorganise some code to make this efficient

## ToDo
- [ ] resolve performance issues when using multi touch

## Please check if the PR fulfills these requirements
- [ ] Self-review: I am confident this is the simplest and clearest way to achieve the expected behaviour
- [ ] There are no dependencies on other PRs or I have linked dependencies through Zenhub
- [ ] The commit messages are clean and understandable
- [ ] Tests for the changes have been added (for bug fixes / features)

janek committed:
initial setup

janek committed:
changed timestamp to Double in Kotlin, removed unused func in GLRenderer

janek committed:
cosmetics - removed some comments, removed a typo

janek committed:
removed some whitespaces

janek committed:
retuned hand-made velocity factor from 0.75 to 0.4

rikner committed:
Merge branch 'master' into improve-touch

rikner committed:
pass timestamp as Long instead of Double, use @_cdecl for onNativeTouch

rikner committed:
move TouchValues data class and MotionEvent extension declaration to top-level

rikner committed:
update SDL (onNativeTouch removed)

rikner committed:
Merge branch 'master' into improve-touch

rikner committed:
cleanup SDLOnTouchListener: rename p to pressure, rename time to timestamp, move MotionEvent to bottom, merge switch cases,

rikner committed:
Merge branch 'master' into improve-touch

rikner committed:
return to SDL master (not removing onNativeTouch)

rikner committed:
rename onNativeTouch to onNativeTouchUIKit to prevent conflict with SDL implementation

rikner committed:
Merge branch 'master' into improve-touch

michaelknoch committed:
check for os(Android) instead of !os(macOS)

Co-Authored-By: rikner <rikner@web.de>

rikner committed:
lint swift files

rikner committed:
Merge branch 'improve-touch' of github.com:flowkey/UIKit-SDL into improve-touch

rikner committed:
refactor UIEvent.description using map()

ephemer committed:
Fix multitouch

rikner committed:
update to SDL version ignoring .DS_Store

rikner committed:
Merge branch 'master' into improve-touch-timestamps

rikner committed:
tune UIScrollView deceleration timing (as seen in b6c6d5a)

rikner committed:
remove UIEvent+fromSDL_Event.swift from xcode project (Android only), lint onNativeTouch

ephemer commented:
@rikner what is the status of this PR? I'm marked to review it but it's missing a description etc. Can you please update it when you get a chance and mark me again for review if it's ready? I'm happy to do some work on this at some point myself too if needed

rikner commented:
@ephemer In the initial PR (#265) we refactored the timestamps and introduced multitouch, but had  performance issues. I split the initial PR into #282 and this one. Currently the performance issues still exist when using multiple fingers. That's the most important issue we should solve.

rikner committed:
Merge branch 'master' into improve-touch-part2

rikner commented:
@ephemer 
@janek and me merged the latest master and ran this on a device to reproduce the performance issue. As far as I remember the player got unresponsive for a while when using multiple fingers, but that's not the case anymore. We also looked at the Profiler in Android Studio and saw that the CPU usage was not different from when using single vs multi-touch. Could it be that this has been solved by some other PR?

ephemer commented:
Hey @rikner thanks for the update here! I would like to try it especially on a device with weak performance to see, it should be immediately obvious whether there are still issues here.

AFAIK we haven't changed anything in another PR so I'd be kind of surprised if the issues have spontaneously gone away. Also curious as to whether you tried this with our app or with the test app (which might hint whether the performance issues are affected by a larger / deeper view hierarchy). Feel free to ping me when you have time and we can have a look at this together

rikner commented:
@ephemer @janek and me also tested it on the rather crappy samsung phone we have in the office. we did not see anything suspicious in the CPU usage. But let's compare the CPU usage between this branch and `master` again. I'll get to you with that.
